tests/test_datasets.py

```diff
@@ -196,38 +203,37 @@ class DatasetRefTestCase(unittest.TestCase):
 
     def setUp(self):
```

Comment:
Are there any tests of components in DatasetRef?

Reply:
There are some indirect ones over in `test_sqlRegistry.py`.

Reply:
Can we have an explicit test here? Especially given the changes to components handling discussed above.

Reply:
Done, and thanks for pushing for this; at first I'd thought the tests even caught a bug in the new code, but instead they caught a bug in the new test code (trying to pass a `DatasetType` instead of a `DatasetRef` as a component) that was worth adding a special check for to yield a better error message.